### PR TITLE
blockchain integration frontend

### PR DESCRIPTION
**Pull request by nicholasmueller**
Issue/Task Number: [711](https://github.com/omisego/ewallet/issues/711) [713](https://github.com/omisego/ewallet/issues/713)  [1120](https://github.com/omisego/ewallet/issues/1120) [714](https://github.com/omisego/ewallet/issues/714)

# Overview
(name of branch is no longer relevant, includes all changes from `frontend-eth` branch)

- blockchain wallets tab to create cold wallet references, balances of blockchain wallets, and relevant transactions with advanced filtering 
- prompt to enable metamask within cold wallet detail to make 'top-up' to hot wallets
- hot wallets can transfer to cold wallets only
- generate deposit addresses for internal wallets
- internal wallets can make internal & external transfers
- new configuration tab layout (blockchain settings not implemented yet)
- new import blockchain token flow

### CODE REVIEW

File: apps/ewallet/lib/ewallet/web/v1/serializers/transaction_serializer.ex [hunk 1]
```diff
@@ -44,6 +44,8 @@ defmodule EWallet.Web.V1.TransactionSerializer do
       id: transaction.id,
       idempotency_token: transaction.idempotency_token,
       blockchain_tx_hash: transaction.blockchain_tx_hash,
+      from_blockchain_address: transaction.from_blockchain_address,
```
[the remark this answers is not in the document]
Comment (nicholasmueller) [reply apps/ewallet/lib/ewallet/web/v1/serializers/transaction_serializer.ex]: so based on what med said ill keep it as is for now

File: apps/frontend/assets/src/omg-blockchain-wallet-modal/index.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  const [ name, setName ] = useState('')
+  const [ address, setAddress ] = useState('')
+  const [ loading, setLoading ] = useState(false)
+
+  const submit = async () => {
+    setLoading(true)
+    await createBlockchainWallet({
+      name,
+      type: 'cold',
+      address: address.toLowerCase()
```
[the remark this answers is not in the document]
Comment (nicholasmueller) [reply apps/frontend/assets/src/omg-blockchain-wallet-modal/index.js]: `blockchainwallet.get_balances` still throws 500 unless I send all lower case... can wait for the backend fix before merging this

[the remark this answers is not in the document]
Comment (mederic-p) [reply apps/frontend/assets/src/omg-blockchain-wallet-modal/index.js]: I believe I fixed it in a pending PR. I'll make a small separate PR to fix it now.

File: apps/frontend/assets/src/omg-create-blockchain-transaction-modal/blockchainTokens.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const blockchainTokens = [
```
[the remark this answers is not in the document]
Comment (nicholasmueller) [reply apps/frontend/assets/src/omg-create-blockchain-transaction-modal/blockchainTokens.js]: actually not using this, was laying around from poc days, will remove

File: apps/frontend/assets/src/omg-import-token-modal/tokenTransferStep.js [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+        />
+        {this.state.step === 1 && (
+          <div style={{ marginBottom: '40px' }}>
+            <StyledSelectInput
+              inputProps={{
+                label: 'Amount to send',
+                value: this.state.fromTokenAmount,
+                onChange: this.onChangeAmount('fromToken'),
+                type: 'amount',
+                maxAmountLength: 18,
```
[the remark this answers is not in the document]
Comment (nicholasmueller) [reply apps/frontend/assets/src/omg-import-token-modal/tokenTransferStep.js]: ah this is just input validation, so can't transfer more than 99,999,999,999,999,999 of whatever token. completely arbitrary

File: apps/frontend/assets/src/omg-page-configuration/BlockchainSettings.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+          {props.enableBlockchain && (
+            <SubSettingContainer>
+              <ConfigRow
+                name={'Network'}
+                value={props.blockchainNetwork}
+                onSelectItem={props.onSelectBlockchainNetwork}
+                onChange={props.onChangeInput('blockchainNetwork')}
+                type='select'
+                options={configurations.blockchain_netowrk
+                  ? configurations.blockchain_network.options.map(option => ({
```
[the remark this answers is not in the document]
Comment (nicholasmueller) [reply apps/frontend/assets/src/omg-page-configuration/BlockchainSettings.js]: fetched from `/configuration.all`. how u set those configs i have no idea

File: apps/frontend/assets/src/omg-internal-to-external-modal/index.js [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+    e.preventDefault()
+    this.setState({ submitting: true })
+    try {
+      const fromAmount = formatAmount(
+        this.state.fromTokenAmount,
+        _.get(this.state.fromTokenSelected, 'token.subunit_to_unit')
+      )
+      const result = await this.props.transfer({
+        fromAddress: this.props.wallet.address,
+        toAddress: this.state.toAddress.trim().toLowerCase(),
```
[the remark this answers is not in the document]
Comment (mederic-p) [reply apps/frontend/assets/src/omg-internal-to-external-modal/index.js]: Shouldn't need to lower case anymore after https://github.com/omisego/ewallet/pull/1161 gets merged

**Pull request merged**